src/test/javascript/spec/service/programming-exercise-grading.service.spec.ts

```diff
@@ -104,8 +113,8 @@ describe('ProgrammingExerciseGradingService', () => {
             .pipe(tap((newTestCases) => (testCasesExercise1 = newTestCases)))
             .subscribe();
 
-        expect(getStub).to.have.been.calledOnce;
-        expect(testCasesExercise1).to.equal(testCases1);
+        expect(getStub).toHaveBeenCalledTimes(1);
+        expect(testCasesExercise1).toBe(testCases1);
```

Comment:
There is some inconsistency in your tests regarding testCases1 and testCases2. In line 90 and 99 you use `toEqual()`, here you use `toBe()`. As these are arrays, I would suggest sticking to `toEqual()`.